lib/pure/terminal.nim

```diff
@@ -468,13 +468,13 @@ proc resetAttributes*(f: File) =
 
 type
   Style* = enum          ## different styles for text output
-    styleBright = 1,     ## bright text
+    styleBright,         ## bright text
```

Comment:
But this is not the same. ``styleBright`` is now 0. Whether that makes a difference for terminal.nim, I don't know. Anyway, an enum not starting with 0, is not an enum with a hole.

Reply:
Sure thing but the inconsistency was bothering me, a quick look shows that nowhere in the code the enum value as an ordinal matters.

Reply:
Oh, styleBright has to be 1, and then the others can follow.

Reply:
It is used here: https://github.com/LemonBoy/Nim/blob/c30612963a4d5f37bc38012e7911e559aa73d8ba/lib/pure/terminal.nim#L503. The current PR will cause all the style encodings to shift by 1. I think I should had a test case or 2 to prevent this in future.

Reply:
> I think I should had a test case or 2 to prevent this in future.

Definitely, heh

> Oh, styleBright has to be 1, and then the others can follow.

Not quite because between blink (5) and reverse (7) there's a unused value.

Reply:
I should have carefully reviewed this (on phone now).. but basically this PR should not change the ordinal values of this enum. *If it's not broken don't fix rule applies here :)*

Reply:
Can this enum not be as is, with holes? If not, I'll find out a sensible name for the missing enum with ordinal 6 when I get to a computer.

Reply:
A quote from the manual:

> The set's basetype can only be an ordinal type of a certain size.

The holes make the enum a non-ordinal type.

> Can this enum not be as is, with holes?

I think you can also do something like this to avoid the dependency on the ordinal value of the elements:

```nim
const styleToSeq: array[styleBright..styleStrikethrough, int] = [ 1, 2, 3, 4, 6, 7, 8 ]
```

Reply:
@LemonBoy That suggestion would work, but would increase the maintenance cost of keeping that enum up to date.. one would need to update the enum and that const.

How how just filling that "hole"? Just add an enum `styleBlinkRapid` **after** `styleBlink`. See https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_(Select_Graphic_Rendition)_parameters.

Reply:
There is no downside to doing so. If someone is formatting their text with that new ANSI code, and if their terminal supports rapid blinking, then that will happen, else no blinking will happen. It's up to the code implementer to decide; terminal.nim is just providing the API.

Reply:
Since this change is orthogonal to the scope of the PR and you're far more familiar with `terminal.nim` I'll just wait for you to patch this first :)

Reply:
@LemonBoy OK, will send a PR in a few mins.

Reply:
Done in https://github.com/nim-lang/Nim/pull/8282. @Araq Can you please review and merge that PR so that this PR can move forward. Thanks!

Reply:
@Lemonboy Can you please rebase this to devel? Now this PR shouldn't need to modify terminal.nim.